src/controllers/trip.controller.js

```diff
@@ -116,6 +116,31 @@ class TripController {
     ResponseService.setSuccess(200, 'List of all available locations', availableLocations);
     return ResponseService.send(res);
   }
+
+  /**
+ *
+ *
+ * @static
+ * @param {req} req
+ * @param {res} res
+ * @returns {response} @memberof Trips
+ */
+  static async editOpenTripRequest(req, res) {
+    const { userData } = req, userId = userData.id, { tripId } = req.params;
+    const findTripRequest = await RequestService.findRequestByProperty({ requesterId: userId }, tripId, { status: !('accepted') });
+    if (findTripRequest) {
+      const [, [{ dataValues }]] = await TripService.updateTrip(
+        { userId, tripId },
+        { ...req.body }
+      );
+      await RequestService.updateRequest({ requesterId: userId, tripId }, { status: 'pending' });
```

Comment:
Don't change the status of the request to `pending`. Instead, check that the request that's being updated is `pending` and open.

Reply:
@adesege i am not clearly getting the meaning here. What i wanted to do is: Once a trip is updated in the Trip table, immediately the status in the Request table will be changed to `pending` for example when it was `rejected`

Reply:
The scope of the task is to allow a requester to update his/her open request. Read the acceptance criteria again for more details.

Reply:
If i get you well, you want me to have 3 statuses `open`, `rejected` and `approved` .  And i have to change where i was using `pending` to `open`?  The idea that bernard said is once the status of a trip request is `pending` or `rejected` thats only when a trip request can be updated

Reply:
When a request is `open`, that means it's still `pending` approval.